scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/SchedulerStateRecoverHelper.java

```diff
@@ -126,37 +129,81 @@ public RecoveredSchedulerState recover(long loadJobPeriod, RMProxy rmProxy) {
         return new RecoveredSchedulerState(pendingJobs, runningJobs, finishedJobs);
     }
 
+    private void handleJob(RMProxy rmProxy, Vector<InternalJob> pendingJobs, Vector<InternalJob> runningJobs,
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Replace the synchronized class "Vector" by an unsynchronized one such as "ArrayList" or "LinkedList". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1149)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Replace the synchronized class "Vector" by an unsynchronized one such as "ArrayList" or "LinkedList". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1149)

Reply:
I do not insist, but why do you use vector?

https://stackoverflow.com/questions/1386275/why-is-java-vector-class-considered-obsolete-or-deprecated

Reply:
As from this PR maybe we could indeed refactor this to a synchronized list?